language-server/src/services/configuration.js

```diff
@@ -81,16 +85,30 @@ export class Configuration {
     });
   }
 
-  /** @type () => Promise<DocumentSettings> */
-  async get() {
+  /** @type (documentUri?: string) => Promise<DocumentSettings> */
+  async get(documentUri) {
     if (!this.#settings) {
-      /** @type unknown */
-      const result = await this.#server.workspace.getConfiguration({
-        section: "jsonSchemaLanguageServer"
-      });
-      const settings = result ?? {};
-      /** @type unknown */
-      const fullSettings = { ...this.#defaultSettings, ...settings };
+      /** @type {unknown[]} */
+      const config = await this.#server.workspace.getConfiguration([
+        { section: "jsonSchemaLanguageServer" },
+        { section: "editor", scopeUri: documentUri },
+        { section: "files.eol" }
+      ]);
+      const [extensionSettings, editorSettings, eol] = /** @type [{ defaultDialect?: string; schemaFilePatterns?: string[] },{ tabSize?: number; insertSpaces?: boolean; detectIndentation?: boolean }, string ] */ (config);
+      /** @type {{ tabSize?: number; insertSpaces?: boolean; detectIndentation?: boolean, endOfLine?: string }} */
```

Comment:
This is too messy. Use `@typedef` to give names to the object types that you can use here.